awx_collection/plugins/modules/tower_notification.py

```diff
@@ -304,125 +325,121 @@
-        recipients=dict(required=False, type='list', elements='str'),
-        use_tls=dict(required=False, type='bool'),
-        host=dict(required=False),
-        use_ssl=dict(required=False, type='bool'),
-        password=dict(required=False, no_log=True),
-        port=dict(required=False, type='int'),
-        channels=dict(required=False, type='list', elements='str'),
-        token=dict(required=False, no_log=True),
-        account_token=dict(required=False, no_log=True),
-        from_number=dict(required=False),
-        to_numbers=dict(required=False, type='list', elements='str'),
-        account_sid=dict(required=False),
-        subdomain=dict(required=False),
-        service_key=dict(required=False, no_log=True),
-        client_name=dict(required=False),
-        message_from=dict(required=False),
-        api_url=dict(required=False),
-        color=dict(required=False, choices=['yellow', 'green', 'red', 'purple', 'gray', 'random']),
-        rooms=dict(required=False, type='list', elements='str'),
-        notify=dict(required=False, type='bool'),
-        url=dict(required=False),
-        headers=dict(required=False, type='dict', default={}),
-        server=dict(required=False),
-        nickname=dict(required=False),
-        targets=dict(required=False, type='list', elements='str'),
+        new_name=dict(),
+        description=dict(),
+        organization=dict(),
+        notification_type=dict(required=True, choices=[
+            'email', 'grafana', 'hipchat', 'irc', 'mattermost',
+            'pagerduty', 'rocketchat', 'slack', 'twilio', 'webhook'
+        ]),
+        notification_configuration=dict(type='dict'),
+        messages=dict(type='dict', default={'started': None, 'success': None, 'error': None, 'workflow_approval': None}),
+        username=dict(),
+        sender=dict(),
+        recipients=dict(type='list', elements='str'),
+        use_tls=dict(type='bool'),
+        host=dict(),
+        use_ssl=dict(type='bool'),
+        password=dict(no_log=True),
+        port=dict(type='int'),
+        channels=dict(type='list', elements='str'),
+        token=dict(no_log=True),
+        account_token=dict(no_log=True),
+        from_number=dict(),
+        to_numbers=dict(type='list', elements='str'),
+        account_sid=dict(),
+        subdomain=dict(),
+        service_key=dict(no_log=True),
+        client_name=dict(),
+        message_from=dict(),
+        api_url=dict(),
+        color=dict(choices=['yellow', 'green', 'red', 'purple', 'gray', 'random']),
+        rooms=dict(type='list', elements='str'),
+        notify=dict(type='bool'),
+        url=dict(),
+        headers=dict(type='dict', default={}),
+        server=dict(),
+        nickname=dict(),
+        targets=dict(type='list', elements='str'),
         state=dict(choices=['present', 'absent'], default='present'),
     )
 
+    # Create a module for ourselves
     module = TowerModule(argument_spec=argument_spec, supports_check_mode=True)
```

Comment:
nit pick as we get more volume of content moving - `supports_check_mode=True` is redundant with this base class, because `supports_check_mode=False` doesn't work. So better to not put it in modules, since it's the domain of the base module class.

https://github.com/ansible/awx/blob/2264a98c045c02203f2a5361130861cca6259ecf/awx_collection/plugins/module_utils/tower_api.py#L58

that's a general issue, but if you're already pushing changes, I bring it up as a might-as-well kind of thing.

Reply:
100%, this should be removed from all of the modules since the module_util deals with this directly. I added a trello card to track this. If you are ok with it I think I'd rather wait for the rest of the module conversions and then push one PR with all of the modules having the same change to them.